src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -138,10 +138,13 @@ internal BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, Bin
             }
 
             Conversion conversion;
+            // Among other things, MakeDeconstructionConversion() will handle
+            // value escape analysis for the Deconstruct() method group.
```

Comment:
note that this escape analysis eventually occurs through the already existing call to `BindMethodGroupInvocation` on the synthesized `Deconstruct` call.